src/search/search.component.ts

```diff
@@ -0,0 +1,220 @@
+	 * @type {string}
+	 * @memberof SearchChange
+	 */
+	value: string;
+}
+
+/**
+ * @export
+ * @class Search
+ * @implements {ControlValueAccessor}
+ */
+@Component({
+	selector: "ibm-search",
+	template: `
+		<div class="bx--form-item">
+			<div
+				class="bx--search"
+				[ngClass]="{
+					'bx--search--sm': size === 'sm',
+					'bx--search--lg': size === 'lg',
+					'bx--search--light': theme === 'light'
+				}"
+				role="search">
+				<label class="bx--label" [for]="id">Search</label>
+				<input
+					class="bx--search-input"
+					type="text"
+					role="search"
+					[id]="id"
+					[value]="value"
+					[placeholder]="placeholder"
+					[disabled]="disabled"
+					[required]="required"
+					[attr.aria-labelledby]="ariaLabelledby"
+					(input)="onSearch($event.target.value)"/>
+				<svg
+					class="bx--search-magnifier"
+					width="16"
+					height="16"
+					viewBox="0 0 16 16">
+					<path
+						d="M6.5 12a5.5 5.5 0 1 0 0-11 5.5 5.5 0 0 0 0 11zm4.936-1.27l4.563 4.557-.707.708-4.563-4.558a6.5 6.5 0 1 1 .707-.707z"
+						fill-rule="nonzero"/>
+				</svg>
+				<button
+					class="bx--search-close"
+					[ngClass]="{
+						'bx--search-close--hidden': !value || value === 0
+					}"
+					title="Clear search input"
+					[attr.aria-label]="ariaLabel"
+					(click)="clearSearch()">
+					<svg
+						width="16"
+						height="16"
+						viewBox="0 0 16 16"
+						xmlns="http://www.w3.org/2000/svg">
+						<path
+							d="M8 6.586L5.879 4.464 4.464 5.88 6.586 8l-2.122 2.121 1.415 1.415L8 9.414l2.121 2.122 1.415-1.415L9.414
+							   8l2.122-2.121-1.415-1.415L8 6.586zM8 16A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"
+							fill-rule="evenodd"/>
+					</svg>
+				</button>
+			</div>
+		</div>
+	`,
+	providers: [
+		{
+			provide: NG_VALUE_ACCESSOR,
+			useExisting: Search,
+			multi: true
+		}
+	]
+})
+export class Search implements ControlValueAccessor {
+	/**
+	 * Variable used for creating unique ids for search components.
+	 */
+	static searchCount = 0;
+
+	/**
+	 * `light` or `dark` search theme.
+	 */
+	@Input() theme: "light" | "dark" = "dark";
+	/**
+	 * Size of the search field.
+	 */
+	@Input() size: "sm" | "lg" = "lg";
+	/**
+	 * Set to `true` for a disabled search input.
+	 */
+	@Input() disabled = false;
+	/**
+	 * Sets the name attribute on the `input` element.
+	 */
+	@Input() name: string;
+	/**
+	 * The unique id for the search component.
+	 */
+	@Input() id = `search-${Search.searchCount}`;
+	/**
+	 * Reflects the required attribute of the `input` element.
+	 */
+	@Input() required: boolean;
+	/**
+	 * Sets the value attribute on the `input` element.
+	 */
+	@Input() value = "";
+	/**
+	 * Sets the placeholder attribute on the  `input` element.
+	 */
+	@Input() placeholder = "Search";
+	/**
+	 * Used to set the `aria-label` attribute on the input element.
+	 */
+	// tslint:disable-next-line:no-input-rename
+	@Input("aria-label") ariaLabel = "Clear search input";
```

Comment:
This (and all other hard-coded default English strings) should be set from a field in `src/i18n/en.json` using `I18n` service.
See `src/loading/loading.component.ts` `title` for example.